master/master.cfg

```diff
@@ -46,32 +46,52 @@ WORKER_SECRET = Path('halide_bb_pass.txt').read_text().strip()
 WEBHOOK_SECRET = Path('webhook_token.txt').read_text().strip()
 WWW_PASSWORD = Path('buildbot_www_pass.txt').read_text().strip()
 
+# SERVER SETTINGS
+
+ARTIFACTS_DIR = os.environ.get('HALIDE_BB_MASTER_ARTIFACTS_DIR', '/home/abadams/artifacts')
+REPO_DIR = Path(__file__, '..', '..').resolve()
+
 # LLVM
 
 # At any given time, we test (at least) 3 LLVM versions:
 # - the current trunk (changes daily)
 # - the most recent release (expected to be stable)
 # - an older release (expected to be stable)
 #
-# the branches that correspond to these will rotate as new versions
-# are released, but the underlying test logic should not need changing
+# The branches that correspond to these will rotate as new versions
+# are released, but the underlying test logic should not need changing.
 
-LLVM_TRUNK_BRANCH = 'main'
-LLVM_RELEASE_BRANCH = 'release/11.x'
-LLVM_OLD_BRANCH = 'release/10.x'
+Version = namedtuple('Version', ['major', 'minor', 'patch'])
+VersionedBranch = namedtuple('VersionedBranch', ['ref', 'version'])
```

Comment:
`ref` seems like an odd name choice, since the git branch name is what is filled in. Why is this a better choice than `branch` or `branch_name`?

Reply:
Because it could be a tag instead (eg. we might add a builder for the release tag `v10.0.0` with just a force scheduler behind it)